src/test/kotlin/org/virtuslab/pulumikotlin/codegen/utils/StringUtilsKtTest.kt

```diff
@@ -27,6 +27,11 @@ internal class StringUtilsKtTest {
         assertEquals("word", "Word".decapitalize())
     }
 
+    @Test
+    fun `should turn the expected number of character lowercase`() {
+        assertEquals("sslCertificate", "SSLCertificate".lowercaseBefore(3))
+    }
```

Comment:
Covering edge cases like `lowercaseBefore(string.length)`, `lowercaseBefore(1)` and `lowercaseBefore(0)` would be nice (and cheap).